diesel/src/query_source/joins.rs

```diff
@@ -180,3 +174,33 @@ impl<DB: Backend> QueryFragment<DB> for LeftOuter {
         Ok(())
     }
 }
+
+use super::{Succ, Never, AppearsInFromClause};
+
+impl<T, Left, Right, Kind> AppearsInFromClause<T> for Join<Left, Right, Kind> where
+    Left: AppearsInFromClause<T>,
+    Right: AppearsInFromClause<T>,
+    Left::Count: Plus<Right::Count>,
```

Comment:
ℹ️ Constraint D: Table `T` occurs on left side of join one more time than on right side

Actually, why is that?